src/turing_machine/machine.py

```diff
@@ -0,0 +1,41 @@
+class TuringMachine:
+    def __init__(self, tape, initial_state, final_states, transition_function):
+        self.tape = list(tape)
+        self.head = 0
+        self.state = initial_state
+        self.final_states = final_states
+        self.transition_function = transition_function
+        self.match_position = -1
+
+    def step(self):
+        current_char = self.tape[self.head]
+        if (self.state, current_char) in self.transition_function:
+            next_state, write_char, direction = self.transition_function[(self.state, current_char)]
+            self.tape[self.head] = write_char
+            self.state = next_state
+            if direction == 'R':
+                self.head += 1
+            elif direction == 'L':
+                self.head -= 1
+            if self.state in self.final_states:
+                self.match_position = self.head
+        else:
+            raise Exception("Error")
+
+    def run(self):
+        while self.state not in self.final_states:
+            self.step()
+
+    def get_tape(self):
+        return ''.join(self.tape)
+
```

Comment:
Zaimplementuj metodę:
```
@abstractmethod
def run_state(self):
    pass
```

To będzie metoda która będzie wykorzystywać wczytany przeze mnie program z pliku konfiguracyjnego i na podstawie aktualnego stanu maszyny będzie decydować o kolejnym stanie oraz operacjacj na taśmie.

Wartością zwracaną przez tą metodę będzie krotka z 3 polami: `(<nowy stan maszyny>, <lista nowych wartości taśm>, <lista operacji na taśmach>)`.

Twoim zadaniem, w metodzie `step` będzie wywołanie tej metody abstrakcyjnej i sprawdzenie czy zwracana przez nią wartość jest zgodna z oczekiwaniem. Jeśli tak, to użyj tych wartości to ustalenie nowego stanu, oraz wykonania odpowiednich operacji na taśmach.